### DISCUSSION TITLE
Anyone else having trouble with SD card writing in 14.0.0.1?

### DISCUSSION THREAD

**Discussion by btsimonh** [arendst/Tasmota#21471] [Q&A]
I'm writing to SD Card every 10 seconds, and sometimes it crashes TAS....
Anyone else had issues with core 3.0.0 and filesystem access?

**Comment by sfromis** [comment]
Debugging details like specific symptoms might be "interesting". Also details like if it is depleting memory over time.

**Reply by btsimonh** [reply]
I'm keeping an eye on it.  It's been up 2 1/2h since last restart after changing the SD card.  
backtrace went back to SPIWriteNL; looking at the code, I guess the data ptr it was trying to write from was invalid.
(from ff_sd_write, originating around fclose() ).
Could be a bad SD card.... but would be nice to think we'd not crash in that circumstance.  But hard to test without a range of known bad SD cards :).

**Reply by btsimonh** [reply]
ok, I have it reproducible on both esp32cam and esp32S3cam.
on S3, it does not crash with the release version 
(tasmota32s3.bin | http://ota.tasmota.com/tasmota32/release/tasmota32s3.bin | 1956k | 20240514 16:12)

but it does crash with the development version:
(tasmota32s3.bin | http://ota.tasmota.com/tasmota32/tasmota32s3.bin | 1973k | 20240522 17:19

I will create an issue with a full description...

**Reply by btsimonh** [reply]
New issue with full reproduction details here:
https://github.com/arendst/Tasmota/issues/21473

**Reply by btsimonh** [reply]
ok, the rebooting was the watchdog.  Turned that off with a define, and it's better.
On esp32S3, I get good performance, and no hanging.
But on esp32cam (ESP Chip Id | 3667140 (ESP32-D0WDQ6 v1.0) ), it randomly stalls the main loop (but reproducibly), and it seems to be after file.open, and during reading and sending the file over http - but I suspect it's a more general SD card thing (as I saw the watchdog trigger inside the close function), indeed possibly hardware related, as we know this hardware is not top quality.

Edit: tested on another ai-thinker style esp32cam, and it's just as fragile.  I may turn the watchdog back on, as at least then when it fails, it reboots.
